### ISSUE TITLE
BUG: Custom class names not persisted in finetuned YOLOX checkpoints

### ISSUE THREAD

**Issue by pxydi** [Libre-YOLO/libreyolo#61]
Hi LibreYOLO team!

**Description**

When finetuning a YOLOX model with custom class names, the checkpoint's names field still contains COCO class names instead of the user-defined ones. This means that after loading a finetuned model, `model.names` maps class IDs to wrong labels unless the user manually overrides it (during inference).

**Steps to reproduce**

1. Provide a [data.yaml](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) with custom class names, e.g.:
```
nc: 3
names: ['my_label_1', 'my_label_2', 'my_label_3']
```

2. Load a pretrained YOLOX backbone and finetune it:
```
model = LibreYOLO("LibreYOLOXs.pt")
modeltrain(data="data.yaml", epochs=50, ...)
```

3. Load the finetunted checkpoint:
```
finetuned = LibreYOLO("runs/train/exp/weights/best.pt")
print(finetuned.names)
```
Expected: {0: 'my_label_1', 1: 'my_label_2', 2: 'my_label_3'}
Actual: {0: 'person', 1: 'bicycle', 2: 'car', ...} (80 COCO names)

**Likely causes**

- `train()` in LibreYOLOX (see models/yolox/model.py) never reads custom names from `data_config`:

```
data_config = load_data_config(data, autodownload=True)
yaml_nc = data_config.get("nc")
# yaml_names = data_config.get("names")  # Never loaded from data_config

if yaml_nc is not None and yaml_nc != self.nb_classes:
    self._rebuild_for_new_classes(yaml_nc)
# self.names is never updated and still holds 80 COCO names
```

- Because `_rebuild_for_new_classes()` in `BaseModel` (see models/base/model.py) never updates `self.names`, `BaseTrainer._save_checkpoint()` saves COCO names into the checkpoint's names field, instead of custom class names. 

- `models/__init__.py` hardcodes `nb_classes=80` when `has_metadata` is `True`:

```
elif has_metadata:
    model = matched_cls(
        model_path=model_path,
        size=size,
        nb_classes=80,   # should use detected nb_classes instead
        device=device,
    )
```

`nb_classes` gets ultimately correctly updated by `_load_weights` later on. However, using directly the detected `nb_classes` would be correct and clearer for the user. 

I can open a PR if it helps.

Kind regards,
Panagiota

**Comment by pxydi** [comment]
I opened this [PR](https://github.com/LibreYOLO/libreyolo/pull/62) if it helps.

KR,

**Comment by brohoya** [comment]
Hi @pxydi!
Good catch, thank you for your fix. Since this also concerns other models, I created the issue https://github.com/LibreYOLO/libreyolo/issues/66
